test/unit/uk/gov/hmrc/apinotificationpull/controllers/NotificationsControllerSpec.scala

```diff
@@ -19,57 +19,96 @@ package uk.gov.hmrc.apinotificationpull.controllers
 import java.util.UUID
 import java.util.concurrent.TimeoutException
 
-import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.{eq => meq, _}
 import org.mockito.Mockito._
+import org.scalatest.BeforeAndAfterEach
 import org.scalatest.mockito.MockitoSugar
 import play.api.http.HeaderNames._
 import play.api.http.Status._
+import play.api.mvc.Results.Ok
 import play.api.test.FakeRequest
+import uk.gov.hmrc.apinotificationpull.connectors.ApiNotificationQueueConnector
 import uk.gov.hmrc.apinotificationpull.fakes.SuccessfulHeaderValidatorFake
-import uk.gov.hmrc.apinotificationpull.model.Notifications
+import uk.gov.hmrc.apinotificationpull.model.{Notification, Notifications}
+import uk.gov.hmrc.apinotificationpull.notifications.NotificationPresenter
 import uk.gov.hmrc.apinotificationpull.services.ApiNotificationQueueService
-import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
 import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
 
 import scala.xml.{Node, Utility, XML}
 import scala.concurrent.Future
 import scala.util.control.NonFatal
 
-class NotificationsControllerSpec extends UnitSpec with WithFakeApplication with MockitoSugar {
-
-  private val notificationId1 = UUID.randomUUID()
-  private val notificationId2 = UUID.randomUUID()
-
-  private val notifications = Notifications(List(s"/notification/$notificationId1", s"/notification/$notificationId2"))
+class NotificationsControllerSpec extends UnitSpec with WithFakeApplication with MockitoSugar with BeforeAndAfterEach{
 
+  implicit val materializer = fakeApplication.materializer
   private val xClientIdHeader = "X-Client-ID"
   private val clientId = "client_id"
 
   private val validHeaders = Seq(ACCEPT -> "application/vnd.hmrc.1.0+xml", xClientIdHeader -> clientId)
+  private val headerValidator = new SuccessfulHeaderValidatorFake
```

Comment:
You could group the relevant variables inside the trait `Setup`, as we did in the other tests.